waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -228,6 +235,13 @@ proc reconnectPeers*(pm: PeerManager, proto: string, backoff: chronos.Duration =
     trace "Reconnecting to peer", peerId=storedInfo.peerId
     discard await pm.dialPeer(storedInfo.peerId, toSeq(storedInfo.addrs), proto)
 
+proc addToBlacklist*(pm: PeerManager, peer: PeerId) = 
+  warn "Adding peer to blacklist"
+  pm.blacklist.add(peer)
```

Comment:
Perhaps this should disconnect the peer and change its `connectedness` state to `ShouldNotConnect`? That way you don't have to keep track of a separate blacklist.